src/worker.h

```diff
@@ -62,7 +59,16 @@ namespace NLV {
       T *child = Nan::ObjectWrap::Unwrap<T>(childObject);
       NLVObjectBasePtr *childPtr = new NLVObjectBasePtr(child);
       child->SetParentReference(childPtr);
-      //parent_->children_.push_back(childPtr);
+      auto parent = ParentClass::Unwrap(parentObject);
+      if(parent) {
+        parent->children_.push_back(childPtr);
+      }
+      Nan::TryCatch try_catch;
+      if(try_catch.HasCaught()) {
```

Comment:
Right, but it needs to actually be defined before the work happens :) In this case I think you need to move it right below the HandleScope above.